python_modules/libraries/dagster-airflow/dagster_airflow/dagster_job_factory.py

```diff
@@ -1,30 +1,173 @@
-from airflow.models.dagbag import DagBag
+from airflow.models.dag import DAG
+from airflow.models.dagrun import DagRun
+from airflow.utils import db
 from dagster import (
-    Definitions,
+    Array,
+    DagsterInvariantViolationError,
+    Field,
+    GraphDefinition,
+    InitResourceContext,
+    JobDefinition,
+    ResourceDefinition,
     _check as check,
 )
+from dagster._core.definitions.utils import validate_tags
+from dagster._core.instance import AIRFLOW_EXECUTION_DATE_STR, IS_AIRFLOW_INGEST_PIPELINE_STR
+from dateutil.parser import parse
 
-from dagster_airflow.dagster_pipeline_factory import (
-    _make_schedules_and_jobs_from_airflow_dag_bag,
-    make_dagster_pipeline_from_airflow_dag,
-    patch_airflow_example_dag,
-)
+from dagster_airflow.airflow_dag_converter import get_graph_definition_args
 from dagster_airflow.utils import (
-    DagsterAirflowError,
+    Locker,
     create_airflow_connections,
+    is_airflow_2_loaded_in_environment,
+    normalized_name,
+    serialize_connections,
 )
 
+# pylint: disable=no-name-in-module,import-error
+if is_airflow_2_loaded_in_environment():
+    from airflow.utils.state import DagRunState
+    from airflow.utils.types import DagRunType
+else:
+    from airflow.utils.state import State
+
+    # pylint: enable=no-name-in-module,import-error
+
+
+def _parse_execution_date_for_job(
+    dag: DAG, run_tags: Mapping[str, str]
+) -> Optional[datetime.datetime]:
+    execution_date_str = run_tags.get(AIRFLOW_EXECUTION_DATE_STR)
+    if not execution_date_str:
+        raise DagsterInvariantViolationError("Expected execution_date_str to be set in run tags.")
+    check.str_param(execution_date_str, "execution_date_str")
+    try:
+        execution_date = parse(execution_date_str)
+        execution_date = execution_date.replace(tzinfo=pytz.timezone(dag.timezone.name))
+    except ValueError:
+        raise DagsterInvariantViolationError(
+            'Could not parse execution_date "{execution_date_str}". Please use datetime'
+            " format compatible with  dateutil.parser.parse.".format(
+                execution_date_str=execution_date_str,
+            )
+        )
+    except OverflowError:
+        raise DagsterInvariantViolationError(
+            'Date "{execution_date_str}" exceeds the largest valid C integer on the system.'.format(
+                execution_date_str=execution_date_str,
+            )
+        )
+    return execution_date
+
+
+def _parse_execution_date_for_asset(
+    dag: DAG, run_tags: Mapping[str, str]
+) -> Optional[datetime.datetime]:
+    execution_date_str = run_tags.get("dagster/partition")
+    if not execution_date_str:
+        raise DagsterInvariantViolationError("dagster/partition is not set")
+    execution_date = parse(execution_date_str)
+    execution_date = execution_date.replace(tzinfo=pytz.timezone(dag.timezone.name))
+    return execution_date
+
+
+def airflow_db(context: InitResourceContext) -> None:
```

Comment:
weird name for function. seems like a grab bag of stateful, side-effect-laden initialization code?